### PR TITLE
[feature] update zero exmaple

### PR DESCRIPTION
**Pull request by ver217**
Update `ZeRO` exmaple with the latest `ShardedModelV2`.

### CODE REVIEW

**Review by FrankLeeeee — APPROVED** [review]

**Comment by FrankLeeeee** [comment]
LGTM.

File: features/zero/train.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    # optimizer
+    # Enable CPU offload for optimizer states
+    optimizer = ShardedOptimizerV2(model, CPUAdam, cpu_offload=True, lr=1e-3)
+    logger.info(f'GPU memory usage after init optim: {torch.cuda.memory_allocated() / 1024**2:.2f} MB', ranks=[0])
+
+    for n in range(NUM_STEPS):
+        # we just use randomly generated data here
+        input_ids, attn_mask = get_data(BATCH_SIZE, SEQ_LEN, VOCAB_SIZE)
+        model.train()
```
Comment (feifeibear) [inline features/zero/train.py:79]: move it out of the loop!

File: features/zero/train.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+from transformers import GPT2Config, GPT2LMHeadModel
+
+
+class GPTLMModel(nn.Module):
+    def __init__(self, hidden_size=768, num_layers=12, num_attention_heads=12, max_seq_len=1024, vocab_size=50257, checkpoint=False):
+        super().__init__()
+        self.checkpoint = checkpoint
+        self.model = GPT2LMHeadModel(GPT2Config(n_embd=hidden_size, n_layer=num_layers,
+                                     n_head=num_attention_heads, n_positions=max_seq_len, n_ctx=max_seq_len, vocab_size=vocab_size))
+        if checkpoint:
```
[the remark this answers is not in the document]
Comment (ver217) [reply features/zero/train.py:20]: DONE

File: features/zero/train.py (hunk 1 above)
Comment (ver217) [reply features/zero/train.py]: DONE

**Pull request merged**